lib/networkaccessmanager.h

```diff
@@ -21,6 +23,10 @@ class NetworkAccessManager : public QNetworkAccessManager {
     /** Get a pointer to the singleton */
     static NetworkAccessManager* instance();
 
+public Q_SLOTS:
+    QStringList supportedSchemesImplementation() const;
+    QUrl urlForRoomEvent(Room *room, const QString &eventId, const QString &mediaId);
+    QUrl urlForFile(Connection *connection, const QString &mediaId);
```

Comment:
I try to use that in a non-QML context (in quotest, actually), and I'm not sure if these two should be in NAM, tbh. One thing that comes to mind is that `Room` already has `urlToDownload()` which extracts the mxc URL from the event - it could be reused to obtain the mxc URL together with the metadata in its fragment. And `urlForFile` really belongs in `Connection`, not here.

Reply:
sure, makes sense

Reply:
Do I understand correctly that you want me to replace `urlToDownload` to return the new url instead?

Reply:
Not quite - I think of having a separate method for now, see `Room/Connection::makeMediaUrl()` in the recent commits.